flink-tests/src/test/java/org/apache/flink/test/checkpointing/RegionFailoverITCase.java

```diff
@@ -97,8 +111,10 @@
 
 	@Before
 	public void setup() throws Exception {
+		HighAvailabilityServicesUtilsTest.TestHAFactory.haServices = new FailHaServices(new TestingCheckpointRecoveryFactory(new FailRecoverCompletedCheckpointStore(1, 1), new StandaloneCheckpointIDCounter()), TestingUtils.defaultExecutor());
 		Configuration configuration = new Configuration();
 		configuration.setString(JobManagerOptions.EXECUTION_FAILOVER_STRATEGY, "region");
+		configuration.setString(HighAvailabilityOptions.HA_MODE, HighAvailabilityServicesUtilsTest.TestHAFactory.class.getName());
```

Comment:
Interesting feature, didn't know about it before. However, I have two questions regarding this integration test:
1.  As before, isn't it enough to use a [RestartStrategy](https://github.com/apache/flink/blob/1dd64058985afd44cae6a5119c4843b2dd85873d/flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/restart/RestartStrategyFactory.java#L136) that returns an exceptionally completed future?
1. We already test that `failGlobal` is invoked from the unit tests. Does the integration test add coverage?

Reply:
Previously, the `RegionFailoverITCase` did not catch bug of [FLINK-13452](https://issues.apache.org/jira/browse/FLINK-13452) due to it did not involve global failover but only region failover. I prefer to add global failover in this integration test to verify the job could be restarted and result still correct.